batched/dense/unit_test/Test_Batched_SerialTbsv.hpp

```diff
@@ -0,0 +1,326 @@
+          h_x1_ref(ib, incx)     = 0.0;
+          h_x1_ref(ib, 2 * incx) = 0.0;
+        } else {
+          h_x_ref(ib, 0)         = 1.0;
+          h_x_ref(ib, 1)         = 0.0;
+          h_x_ref(ib, 2)         = 0.0;
+          h_x1_ref(ib, 0)        = 1.0;
+          h_x1_ref(ib, incx)     = 0.0;
+          h_x1_ref(ib, 2 * incx) = 0.0;
+        }
+      }
+    } else {
+      if constexpr (std::is_same_v<typename ParamTagType::trans,
+                                   Trans::NoTranspose>) {
+        if constexpr (std::is_same_v<typename ParamTagType::diag,
+                                     Diag::NonUnit>) {
+          h_x_ref(ib, 0)         = 1.0;
+          h_x_ref(ib, 1)         = -1.0 / 2.0;
+          h_x_ref(ib, 2)         = -1.0 / 6.0;
+          h_x1_ref(ib, 0)        = 1.0;
+          h_x1_ref(ib, incx)     = -1.0 / 2.0;
+          h_x1_ref(ib, 2 * incx) = -1.0 / 6.0;
+        } else {
+          h_x_ref(ib, 0)         = 1.0;
+          h_x_ref(ib, 1)         = -1.0;
+          h_x_ref(ib, 2)         = 1.0;
+          h_x1_ref(ib, 0)        = 1.0;
+          h_x1_ref(ib, incx)     = -1.0;
+          h_x1_ref(ib, 2 * incx) = 1.0;
+        }
+      } else {
+        if constexpr (std::is_same_v<typename ParamTagType::diag,
+                                     Diag::NonUnit>) {
+          h_x_ref(ib, 0)         = 0.0;
+          h_x_ref(ib, 1)         = 0.0;
+          h_x_ref(ib, 2)         = 1.0 / 3.0;
+          h_x1_ref(ib, 0)        = 0.0;
+          h_x1_ref(ib, incx)     = 0.0;
+          h_x1_ref(ib, 2 * incx) = 1.0 / 3.0;
+        } else {
+          h_x_ref(ib, 0)         = 2.0;
+          h_x_ref(ib, 1)         = -2.0;
+          h_x_ref(ib, 2)         = 1.0;
+          h_x1_ref(ib, 0)        = 2.0;
+          h_x1_ref(ib, incx)     = -2.0;
+          h_x1_ref(ib, 2 * incx) = 1.0;
+        }
+      }
+    }
+  }
+
+  Kokkos::fence();
+
+  Kokkos::deep_copy(ref, h_ref);
+  Kokkos::deep_copy(x0, h_x0);
+  Kokkos::deep_copy(x1, h_x1);
+  Kokkos::deep_copy(x_ref, h_x_ref);
+  Kokkos::deep_copy(x1_ref, h_x1_ref);
+
+  // Create triangluar or banded matrix
+  create_banded_triangular_matrix<View3DType, View3DType,
+                                  typename ParamTagType::uplo>(ref, A, k,
+                                                               false);
+  create_banded_triangular_matrix<View3DType, View3DType,
+                                  typename ParamTagType::uplo>(ref, Ab, k,
+                                                               true);
+
+  // tbsv
+  Functor_BatchedSerialTbsv<DeviceType, View3DType, View2DType, ParamTagType,
+                            AlgoTagType>(Ab, x0, k, 1)
+      .run();
+
+  // tbsv with incx == 2
+  Functor_BatchedSerialTbsv<DeviceType, View3DType, View2DType, ParamTagType,
+                            AlgoTagType>(Ab, x1, k, incx)
+      .run();
+
+  // Check x0 = x_ref
+  EXPECT_TRUE(allclose<execution_space>(x0, x_ref, 1.e-5, 1.e-12));
```

Comment:
This is a common mistake: you should not pick arbitrary tolerances here you should at least use something like `Kokkos::ArithTraits<ScalarType>::eps()` so it works for both `double` and `float`.
In practice you want to figure out how many operations are performed to compute each entry in x and use that number to estimate how many ulp you have accumulated, then multiply epsilon by that number. Finally that check should be done against a relative error so there is no scaling issues with the tolerance.

Reply:
Sure. I will do that.